spot_wrapper/wrapper.py

```diff
@@ -648,255 +439,298 @@ def __init__(
-                self._estop_client, self._logger, 20.0, self
-            )
-            self._world_objects_task = AsyncWorldObjects(
-                self._world_objects_client,
-                self._logger,
-                10.0,
-                self._callbacks.get("world_objects", None),
-            )
+        # Core Async Tasks
+        self._async_task_list = []
+        self._robot_state_task = AsyncRobotState(
+            self._robot_state_client,
+            self._logger,
+            max(0.0, self._rates.get("robot_state", 0.0)),
+            self._callbacks.get("robot_state", None),
+        )
+        self._robot_metrics_task = AsyncMetrics(
+            self._robot_state_client,
+            self._logger,
+            max(0.0, self._rates.get("metrics", 0.0)),
+            self._callbacks.get("metrics", None),
+        )
+        self._lease_task = AsyncLease(
+            self._lease_client,
+            self._logger,
+            max(0.0, self._rates.get("lease", 0.0)),
+            self._callbacks.get("lease", None),
+        )
+        self._idle_task = AsyncIdle(
+            self._robot_command_client, self._logger, 10.0, self
+        )
+        self._estop_monitor = AsyncEStopMonitor(
+            self._estop_client, self._logger, 20.0, self
+        )
 
-            self._estop_endpoint = None
-            self._estop_keepalive = None
+        self._estop_endpoint = None
+        self._estop_keepalive = None
+        self._robot_id = None
+        self._lease = None
+
+        robot_tasks = [
+            self._robot_state_task,
+            self._robot_metrics_task,
+            self._lease_task,
+            self._idle_task,
+            self._estop_monitor,
+        ]
 
-            robot_tasks = [
-                self._robot_state_task,
-                self._robot_metrics_task,
-                self._lease_task,
-                self._front_image_task,
-                self._idle_task,
-                self._estop_monitor,
-                self._world_objects_task,
-            ]
-
-            if self._point_cloud_client:
-                self._point_cloud_task = AsyncPointCloudService(
-                    self._point_cloud_client,
-                    self._logger,
-                    max(0.0, self._rates.get("point_cloud", 0.0)),
-                    self._callbacks.get("lidar_points", None),
-                    self._point_cloud_requests,
-                )
-                robot_tasks.append(self._point_cloud_task)
+        # Create component objects for different functionality
+        self._robot_params = {
+            "is_standing": self._is_standing,
+            "is_sitting": self._is_sitting,
+            "is_moving": self._is_moving,
+            "at_goal": self._at_goal,
+            "near_goal": self._near_goal,
+            "robot_id": self._robot_id,
+            "estop_timeout": self._estop_timeout,
+            "rates": self._rates,
+            "callbacks": self._callbacks,
+        }
```

Comment:
This should probably also be an object rather than a dict, again something to consider later.

Reply:
Agreed, this works for now (since almost everything is an object in Python), but it would be nice to be explicit about what states are shared at runtime before going to operation.